client/pages/groups.jsx

```diff
@@ -30,22 +35,101 @@ export default class Groups extends React.Component {
         h1: '',
         h2: ''
       }
-
     };
+    this.handleGroupSave = this.handleGroupSave.bind(this);
   }
 
   componentDidMount() {
+
     window.addEventListener('hashchange', event => {
       const newRoute = parseRoute(window.location.hash);
       this.setState({
         route: newRoute,
         groupCount: 0
       });
     });
+    fetch('/api/teams')
+      .then(response => response.json())
+      .then(teamData => {
+        this.setState({
+          teams: teamData
+        });
+      })
+      .catch(error => {
+        console.error('error:', error);
+      });
+
+    const updateEditState = JSON.parse(window.localStorage.getItem('editing-state'));
+    this.setState({
+      isEditing: updateEditState
+    });
+
+    const updateBracketState = JSON.parse(window.localStorage.getItem('brackets-state'));
+    this.setState({
+      brackets: updateBracketState
+    });
+
+    const updateGroupsState = JSON.parse(window.localStorage.getItem('groupStage-state'));
+    this.setState({
+      groupStage: updateGroupsState
+    });
   }
 
-  teamSelected(teamId, event) {
+  handleGroupSave() {
+
+    fetch('/api/brackets', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
```

Comment:
Be sure to send your token along with your request.

Reply:
done.